examples/CMakeLists.txt

```diff
@@ -46,105 +46,6 @@ if(DEAL_II_COMPONENT_EXAMPLES)
-      set(_setup FALSE)
-      if(EXISTS "${_directory}/CMakeLists.txt")
-        file(STRINGS "${_directory}/CMakeLists.txt" _dependency_string
-          REGEX "^if.*DEAL_II.* # keep in one line$"
-          )
-        string(REPLACE "if(" "" _dependency_string "${_dependency_string}")
-        string(REPLACE ") # keep in one line" "" _dependency_string "${_dependency_string}")
-        if("${_dependency_string}" STREQUAL "")
-          set(_setup TRUE)
-        else()
-          # if the dependency string evaluates to TRUE then the example
-          # CMakeLists.txt encounters a fatal error - we want the opposite logic
-          # here so add a NOT.
-          evaluate_expression("
-            if(NOT (${_dependency_string}))
-              set(_setup TRUE)
-            endif()")
-        endif()
-      endif()
-
-      if(_setup)
-        foreach(_build ${DEAL_II_BUILD_TYPES})
-          string(TOLOWER ${_build} _build_lowercase)
-          set(_target "example_${_name}_${_build_lowercase}")
-          string(REPLACE "-" "_" _target "${_target}")
-          add_executable(${_target} ${_step})
-          insource_setup_target(${_target} ${_build})
-          set_target_properties(${_target}
-            PROPERTIES
-            RUNTIME_OUTPUT_NAME "${_name}.${_build_lowercase}"
-            RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/${DEAL_II_EXECUTABLE_RELDIR}"
-            )
-          #
-          # In case CMake is instructed to add rpaths to the library and
-          # exectuble on installation, make sure that we add an additional
-          # rpath to the library location as well:
-          #
-          if(CMAKE_INSTALL_RPATH_USE_LINK_PATH)
-            set_target_properties(${_target}
-              PROPERTIES
-              INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/${DEAL_II_LIBRARY_RELDIR}"
-              )
-          endif()
-
-          add_dependencies(examples ${_target})
-          install(TARGETS ${_target}
-            DESTINATION ${DEAL_II_EXAMPLES_RELDIR}/${_name}
-            )
-        endforeach()
-
-      else()
-
-        message(STATUS "  ${_name} - dependencies not satisfied")
-      endif()
-
-    endforeach()
-
-    #
-    # Also compile some documentation examples:
-    #
-
-    file(GLOB _steps ${CMAKE_CURRENT_SOURCE_DIR}/doxygen/*.cc)
-    foreach(_step ${_steps})
-      get_filename_component(_name ${_step} NAME_WE)
-
-        foreach(_build ${DEAL_II_BUILD_TYPES})
-          string(TOLOWER ${_build} _build_lowercase)
-          set(_target "example_${_name}_${_build_lowercase}")
-          string(REPLACE "-" "_" _target "${_target}")
-          add_executable(${_target} ${_step})
-          insource_setup_target(${_target} ${_build})
-          set_target_properties(${_target}
-            PROPERTIES
-            RUNTIME_OUTPUT_NAME "${_name}.${_build_lowercase}"
-            RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/${DEAL_II_EXECUTABLE_RELDIR}"
-            )
-        endforeach()
-    endforeach()
```

Comment:
This is the part I was talking about.

Reply:
Let's not worry about this. We should convert these code snippets into full tests, or full example steps.